core/state/state_object.go

```diff
@@ -263,9 +262,20 @@ func (self *StateObject) Code(db trie.Database) []byte {
 	return code
 }
 
+// CodeSize returns the size of the contract code associated with this object.
+func (self *StateObject) CodeSize(db trie.Database) int {
+	if self.data.codeSize == nil {
+		self.data.codeSize = new(int)
```

Comment:
I usually prefer using the default value to mean non-initialized. Using -1 would require taking extra care to always initialize it properly (e.g. there is code that just did `Account{}`, which would be invalid from now on).